### PR TITLE
keep null row and remove dw_int64 column

### PR DESCRIPTION
**Pull request by valiantljk**
Fixed Issue 1: a new dw_last_update_int64 column is added for filtering, but need to be dropped 
Fixed issue 2: dw_last_update could have null value, avoid dropping it during filtering

### CODE REVIEW

File: deltacat/compute/compactor/steps/repartition.py [hunk 1]
```diff
@@ -117,12 +118,18 @@ def repartition_range(
...
+
     partition_table_length = 0
     # After re-grouping the tables by specified ranges, for each group, we need concat and stage the tables
     partition_deltas: List[Delta] = []
     for partition_tables in partitioned_tables_list:
         if len(partition_tables) > 0:
-            partition_table: pa.Table = pa.concat_tables(partition_tables)
+            partition_table: pa.Table = pa.concat_tables(
+                partition_tables
+            ).remove_column(0)
```
Comment (raghumdani) [inline deltacat/compute/compactor/steps/repartition.py:131]: should we use `drop` method with actual column name created above? Just to make sure we don't drop any actual column from table?

**Review by raghumdani — COMMENTED** [review]
Thanks for the fix.

File: deltacat/compute/compactor/steps/repartition.py (hunk 1 above)
Comment (valiantljk) [reply deltacat/compute/compactor/steps/repartition.py:131]: good idea

Comment (valiantljk) [reply deltacat/compute/compactor/steps/repartition.py:131]: turned out drop_columns/drop is not supported in our current version 10.0.1; I'll just use remove_column but by first locating the index with the column name;

Comment (raghumdani) [reply deltacat/compute/compactor/steps/repartition.py]: Then how does this line work? https://github.com/ray-project/deltacat/blob/037b584d7b7330dca6be94adaf279ff01593fee8/deltacat/compute/compactor/steps/hash_bucket.py#L59

Comment (valiantljk) [reply deltacat/compute/compactor/steps/repartition.py]: updated.

**Review by raghumdani — APPROVED** [review]
looks good. Please check why the checks are failing.

**Pull request merged**